File: benchmarks/lua/closure_heavy.lua
local function makeAdder(x)
  return function(y)
    return x + y
  end
end

local function closureHeavy(n)
  local closures = {}
  for i = 0, n - 1 do
    closures[i + 1] = makeAdder(i)
  end

  local sum = 0
  for i = 0, n - 1 do
    sum = sum + closures[i + 1](i)
  end
  return sum
end

local n = tonumber(arg[1]) or 0
print(closureHeavy(n))

File: benchmarks/js/closure_heavy.js
function makeAdder(x) {
  return function(y) {
    return x + y;
  };
}

function closureHeavy(n) {
  const closures = [];
  for (let i = 0; i < n; i++) {
    closures.push(makeAdder(i));
  }

  let sum = 0;
  for (let i = 0; i < n; i++) {
    sum += closures[i](i);
  }
  return sum;
}

const n = process.argv[2] ? parseInt(process.argv[2]) : 0;
console.log(closureHeavy(n));
